unprivileged/ubuntu/no-nap/Dockerfile

```diff
@@ -0,0 +1,101 @@
+
+# NGXIN Plus release e.g 23
+ARG NGINX_PLUS_VERSION=23
+
+# Download certificate (nginx-repo.crt) and key (nginx-repo.key) from the customer portal (https://cs.nginx.com)
+# and copy to the build context
+COPY nginx-repo.* /etc/ssl/nginx/
+COPY nginx-plus-api.conf /etc/nginx/conf.d/
+COPY entrypoint.sh /
+
+# Install NGINX Plus
+RUN set -ex \
+  && apt-get update \
+  && apt-get install --no-install-recommends --no-install-suggests -y \
+  curl \
+  sudo\
+  vim\
+  procps\
+  gnupg\
+  apt-utils\
+  apt-transport-https\
+  ca-certificates\
+  distro-info-data\
+  libmpdec2\
+  lsb-release \
+  binutils \
+  net-tools \
+  && apt-get upgrade -y \
+  && \
+  NGINX_GPGKEY=573BFD6B3D8FBC641079A6ABABF5BD827BD9BF62; \
+  found=''; \
+  for server in \
+    ha.pool.sks-keyservers.net \
+    hkp://keyserver.ubuntu.com:80 \
+    hkp://p80.pool.sks-keyservers.net:80 \
+    pgp.mit.edu \
+  ; do \
+    echo "Fetching GPG key $NGINX_GPGKEY from $server"; \
+    apt-key adv --keyserver "$server" --keyserver-options timeout=10 --recv-keys "$NGINX_GPGKEY" && found=yes && break; \
+  done; \
+  test -z "$found" && echo >&2 "error: failed to fetch GPG key $NGINX_GPGKEY" && exit 1; \
+  echo "Acquire::https::plus-pkgs.nginx.com::Verify-Peer \"true\";" >> /etc/apt/apt.conf.d/90nginx \
+  && echo "Acquire::https::plus-pkgs.nginx.com::Verify-Host \"true\";" >> /etc/apt/apt.conf.d/90nginx \
+  && echo "Acquire::https::plus-pkgs.nginx.com::SslCert     \"/etc/ssl/nginx/nginx-repo.crt\";" >> /etc/apt/apt.conf.d/90nginx \
+  && echo "Acquire::https::plus-pkgs.nginx.com::SslKey      \"/etc/ssl/nginx/nginx-repo.key\";" >> /etc/apt/apt.conf.d/90nginx \
+  && printf "deb https://plus-pkgs.nginx.com/ubuntu $(lsb_release -cs) nginx-plus\n" > /etc/apt/sources.list.d/nginx-plus.list \
+  # NGINX Javascript module needed for APIM
+  && apt-get update && apt-get install -y nginx-plus=${NGINX_PLUS_VERSION}* nginx-plus-module-njs=${NGINX_PLUS_VERSION}*  \
+  && rm -rf /var/lib/apt/lists/* \
+  # Install Controller Agent
+  && curl -k -sS -L ${CONTROLLER_URL} > install.sh \
+  && sed -i 's/^assume_yes=""/assume_yes="-y"/' install.sh \
+  && sh ./install.sh -y \
+  # cleanup sensitive nginx-plus data
+  && rm /etc/ssl/nginx/nginx-repo.* \
+  && rm /etc/apt/sources.list.d/nginx-plus.list \
+  && rm /etc/apt/apt.conf.d/90nginx \
+  && apt-key del "$NGINX_GPGKEY"
+
+# Forward request logs to Docker log collector
+RUN ln -sf /proc/1/fd/1 /var/log/nginx-controller/agent.log \
+  && ln -sf /proc/1/fd/2 /var/log/nginx/error.log
+
+# Update ownership for the necessary filesystem objects for running under "nginx" user
+RUN chown -R nginx:nginx /var/run/ \
+  && chown -R nginx:nginx /var/log/nginx/ \
+  && chown -R nginx:nginx /var/log/nginx-controller/ \
+  && chown -R nginx:nginx /var/cache/nginx/ \
+  && chown -R nginx:nginx /etc/controller-agent/ \
+  && chown -R nginx:nginx /etc/nginx/ \
+  && sed -i '/user  nginx;/d' /etc/nginx/nginx.conf
+
+USER nginx
+
+EXPOSE 80
```

Comment:
Seems better for this to be invocation driven than hardcoded?

Reply:
that's actually the plan. This PR is a WIP for testing purposes, just updated the title.